### PR TITLE
Product List, Checkout and Blog enhancement

### PR DESCRIPTION
**Pull request by mirafedas**
- Added sorting by price range in Product List;
- Added input validation and notifications in Checkout component;
- Adjusted styles on the Blog Page.

### CODE REVIEW

File: src/components/Checkout/BillingInfo/BillingInfo.js [hunk 1]
```diff
@@ -11,102 +12,158 @@ const CN = 'billing-info';
...
+
+  validate() {
+    const { showMessage } = this.props;
+
+    if (!this.form.current.reportValidity()) {
+      showMessage(VALIDATION_FAILED);
+      return this.form.current.reportValidity();
+    }
+
+    return this.form.current.reportValidity();
```
Comment (nadiiaLebedenko) [inline src/components/Checkout/BillingInfo/BillingInfo.js]: You can move this call to the separate variable to call this only once. Plus line 28 is redundant.

File: src/components/Checkout/LoginRegister/LoginRegister.js [hunk 2]
```diff
@@ -2,56 +2,106 @@ import React from 'react';
...
+
+  validate() {
+    const { showMessage } = this.props;
+
+    if (!this.form.current.reportValidity()) {
+      showMessage(VALIDATION_FAILED);
+      return this.form.current.reportValidity();
+    }
+
+    return this.form.current.reportValidity();
```
Comment (nadiiaLebedenko) [inline src/components/Checkout/LoginRegister/LoginRegister.js]: See the comment above

File: src/components/Checkout/ShippingMethod/ShippingMethod.js [hunk 3]
```diff
@@ -17,6 +17,7 @@ const ShippingMethod = () => (
           type="radio"
           name="ship-to-this-address"
           value="ship-to-this-address"
+          checked
```
Comment (nadiiaLebedenko) [inline src/components/Checkout/ShippingMethod/ShippingMethod.js:20]: Should it be checked always?

**Review by nadiiaLebedenko — CHANGES_REQUESTED** [review]

File: src/components/Checkout/ShippingMethod/ShippingMethod.js (hunk 3 above)
Comment (mirafedas) [reply src/components/Checkout/ShippingMethod/ShippingMethod.js:20]: Not always, only by default.

Comment (nadiiaLebedenko) [reply src/components/Checkout/ShippingMethod/ShippingMethod.js:20]: Won't it be always checked on rerender?

File: src/components/Checkout/BillingInfo/BillingInfo.js [hunk 4]
```diff
@@ -22,13 +22,14 @@ class BillingInfo extends React.Component {
 
   validate() {
     const { showMessage } = this.props;
+    const isInputValid = this.form.current.reportValidity();
 
-    if (!this.form.current.reportValidity()) {
+    if (!isInputValid) {
       showMessage(VALIDATION_FAILED);
-      return this.form.current.reportValidity();
+      return isInputValid;
```
Comment (nadiiaLebedenko) [inline src/components/Checkout/BillingInfo/BillingInfo.js]: You can easily loose this line, it doesn't have any purpose

File: src/components/Checkout/LoginRegister/LoginRegister.js [hunk 5]
```diff
@@ -18,13 +18,14 @@ class LoginRegister extends React.Component {
 
   validate() {
     const { showMessage } = this.props;
+    const isInputValid = this.form.current.reportValidity();
 
-    if (!this.form.current.reportValidity()) {
+    if (!isInputValid) {
       showMessage(VALIDATION_FAILED);
-      return this.form.current.reportValidity();
+      return isInputValid;
```
Comment (nadiiaLebedenko) [inline src/components/Checkout/LoginRegister/LoginRegister.js]: Same as above

**Review by nadiiaLebedenko — CHANGES_REQUESTED** [review]

**Review by nadiiaLebedenko — APPROVED** [review]

**Pull request merged**